### ISSUE TITLE
Documentation: inline code with links split into multiple blocks

### ISSUE THREAD

**Issue by tspiteri** [rust-lang/rust#83997]
**EDIT(camelid): ⚠️ Please do not open a PR for this! It is still under discussion.**

-----

Right now in the standard library documentation, when a piece of inline code has part of it linked, the code is split into multiple `<code>` blocks, which renders poorly. For example the [`Arc` thread safety section](https://doc.rust-lang.org/nightly/std/sync/struct.Arc.html#thread-safety) contains

![image](https://user-images.githubusercontent.com/18604588/114037935-b6340e00-9881-11eb-9739-0b5b62b03158.png)

Here, between `Arc<` and `RefCell`, and between `<T>` and `>`, there is extra spacing. This is because the doc comment contains
```rust
/// `Arc<`[`RefCell<T>`]`>`
```
which is rendered using three `<code>` blocks as
```html
<code>Arc&lt;</code><a href="../../std/cell/struct.RefCell.html"><code>RefCell&lt;T&gt;</code></a><code>&gt;</code>
```

I think better would be
```rust
/// <code>Arc&lt;[RefCell&lt;T&gt;][RefCell]&gt;</code>
```
which presumably is rendered in one block as
```html
<code>Arc&lt;<a href="../../std/cell/struct.RefCell.html">RefCell&lt;T&gt;</a>&gt;</code>
```
(I'm not sure if the link to `RefCell` gets a `title` attribute as well).

The downside is that now the doc comment markdown is a bit hairy with explicit `<code>`, `&lt;` and `&gt;` spread throughout.

Is this substitution considered an improvement? If so I can grep through the code, make some substitutions, and open a PR.

An alternative solution would be to change rustdoc itself to combine consecutive code blocks, some of which could be wrapped inside `<a>`/`</a>`, into a single block, though I do not have the know-how to open such a PR. I don't even know if there is some downside to this that I am missing, where `<code>` blocks are meant to be separate, maybe for some markdown compliance or something.

**Comment by jyn514** [comment]
Yes, I think this is a good change - one like it has already been accepted (https://github.com/rust-lang/rust/pull/80720) and I need to merge the other, thanks for reminding me (https://github.com/rust-lang/rust/pull/80733). I need to open an issue for rustdoc to do this automatically - I don't know if it's worth spending time to convert the whole standard library by hand in the meantime.

> An alternative solution would be to change rustdoc itself to combine consecutive code blocks, some of which could be wrapped inside <a>/</a>, into a single block, though I do not have the know-how to open such a PR. I don't even know if there is some downside to this that I am missing, where <code> blocks are meant to be separate, maybe for some markdown compliance or something.

See https://github.com/rust-lang/rust/pull/80733#issuecomment-770297395 which I think is a very good approach and we might be able to do with CSS only (cc @camelid).

**Comment by camelid** [comment]
I think we should open a separate issue for changing rustdoc's behavior so it does this automatically, and then close this one. What do you think?

**Comment by camelid** [comment]
Okay, I opened another issue: #84043. I'm not going to close this one without hearing back from other people on what they think, but I added a warning to the top of this issue so someone doesn't put energy into a PR before it is decided what we want to do.

**Comment by notriddle** [comment]
Resolved by https://github.com/rust-lang/rust-clippy/pull/14121
